charm/src/actions.py

```
@@ -84,8 +84,17 @@ def _allow_domains(self, event: ops.ActionEvent) -> None:
             event: The event fired by the action.
         """
         username = event.params["username"]
-        domains = event.params["domains"].split(",")
-        self._execute_command(["allow_domains", username] + domains, event)
+        domains = event.params.get("domains", None)
+        subdomains = event.params.get("subdomains", None)
```

Are we sure domains and subdomains do not contains invalid characters for a shell command at this stage?
If not, we should have some validation.

---

Ok, I see now that we handle the username case in the Django app. I think it would make sense to fail here when the user is not there since we do it for `domain` and `subdomain`. I like failing early.
WDYT?